app/src/main/java/com/victor/adasfeed/ProfileActivity.kt

```diff
@@ -20,47 +20,118 @@ class ProfileActivity : AppCompatActivity() {
+    private lateinit var buttonSave: FloatingActionButton
+    private lateinit var buttonReturnToFeed: Button
+    private var user: User? = null
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_profile)
         Log.d("contexto PA", applicationContext.toString())
         Log.d("ciclo de vida PA", "onCreate")
 
-        textUserName = findViewById(R.id.textUserName)
-        textNickname = findViewById(R.id.textNickname)
-        buttonCall = findViewById(R.id.buttonContact)
-        imageUser = findViewById(R.id.imageUser)
-
+        initViews()
+        // criar os listeners
         val extras = intent.extras
         var uri = Uri.parse("")
-
-        // envia por intents - putExtra
-        // recebe por Bundle
         extras?.let { bundle ->
-            val user = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
-                bundle.getParcelable(EXTRA_KEY, User::class.java)
-            } else {
-                bundle.getParcelable(EXTRA_KEY) as? User
-            }
-            // mais de um parametro na string
-//            textUserName.text = getString(R.string.profile_name, safeUser.userName, safeUser.userNickname)
+            user = extractUser(EXTRA_KEY, bundle)
 
             user?.let { safeUser ->
                 textUserName.text = getString(R.string.profile_name, safeUser.userName)
                 textNickname.text = safeUser.userNickname
                 imageUser.setImageResource(safeUser.imageUser)
+
                 uri = safeUser.tel?.let {
                     Uri.parse("tel:${safeUser.tel}")
                 } ?: Uri.parse("tel:")
             }
         }
+        setupClickListeners(uri)
+    }
+
+    fun updateUserName(newName: String) {
+        textUserName.text = newName
+    }
+
+    fun updateNickname(newNickname: String) {
+        textNickname.text = newNickname
+    }
+
+    private fun setupClickListeners(uri: Uri) {
 
         buttonCall.setOnClickListener {
             val intent = Intent(Intent.ACTION_DIAL, uri)
             startActivity(intent)
             finish()
         }
+
+        userNameField.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+                // This method is called before the text is changed
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                // This method is called when the text is changed
+                val newName = "Nome: " + s.toString()
+
+                // Update the name on the view in real-time
+                updateUserName(newName)
+            }
+
+            override fun afterTextChanged(s: Editable?) {
+                // This method is called after the text is changed
+            }
+        })
+
+        nickNameField.addTextChangedListener(object : TextWatcher {
```

Comment:
The method beforeTextChanged and afteTextChanged are unusable. You can use onTextChanged, for example: `nickNameField.addTextChangedListener{ editable ->
}`